results-loader/src/utils/results-utils.ts

```diff
@@ -43,3 +44,28 @@ export const getFirebaseJwt = (portalUrl: string, portalAccessToken: string, fir
     .then(response => response.json())
     .then(json => json.token);
 };
+
+interface IGetCredentials {
+  resource: Resource;
+  firebaseJwt?: string;
+  tokenServiceEnv: EnvironmentName;
+}
+export const getCredentials = async ({resource, firebaseJwt,
+  tokenServiceEnv}: IGetCredentials): Promise<Credentials> => {
+
+  // The jwt will be ignored if there is a readWriteToken on the resource
+  const client = new TokenServiceClient({ env: tokenServiceEnv, jwt: firebaseJwt });
+  const readWriteToken = client.getReadWriteToken(resource) || "";
+
+  return readWriteToken ? await client.getCredentials(resource.id, readWriteToken) :  await client.getCredentials(resource.id);
+};
```

Comment:
You can leave out the readWriteToken stuff here. The results-loader doesn't need to support them. 
Also I think an async function can return a promise without waiting for it, so the `await` can be dropped in the return statement. I'll be curious if typescript handles this correctly. If it does the return type of this function shouldn't need to be changed.
 
```suggestion
  return client.getCredentials(resource.id);
};
```

Reply:
Fixed.